.github/workflows/build_package.yml

```diff
@@ -149,23 +166,43 @@ jobs:
       # is not enough to just invoke it directly because then pip managed
       # console scripts (like cmake, ninja) will not be on the path.
       ##########################################################################
-      - name: Main distribution (Linux)
-        if: "matrix.build-package == 'main-dist-linux'"
+      - name: Main distribution (Linux-x86_64)
+        if: "matrix.build-package == 'main-dist-linux' && matrix.build-family == 'linux-x86_64'"
         shell: bash
         run: |
           docker run --rm -w=/work \
             -v $PWD:/work \
             "${MANYLINUX_X86_64_IMAGE}" \
             bash -c 'export PATH=/opt/python/cp39-cp39/bin:$PATH; python ./c/build_tools/github_actions/build_dist.py main-dist'
 
+      - name: Main distribution (Linux-AArch64)
+        if: "matrix.build-package == 'main-dist-linux' && matrix.build-family == 'linux-aarch64'"
+        shell: bash
+        run: |
+          docker run --rm -w=/work \
+            -v $PWD:/work \
+            "${MANYLINUX_AARCH64_IMAGE}" \
+            bash -c 'export PATH=/opt/python/cp39-cp39/bin:$PATH; python ./c/build_tools/github_actions/build_dist.py main-dist'
+
       ##########################################################################
       # py-runtime-pkg
       # Builds the iree-runtime wheels.
-      # One step per OS.
+      # One step per OS (and CPU architecture for Linux).
       ##########################################################################
 
-      - name: Build runtime wheels (Linux)
-        if: "matrix.build-package == 'py-runtime-pkg' && matrix.build-family == 'linux'"
+      - name: Build runtime wheels (Linux-x86_64)
+        if: "matrix.build-package == 'py-runtime-pkg' && matrix.build-family == 'linux-x86_64'"
```

Comment:
As these steps are identical, you should be able to omit the extra steps and instead check if the "build family" includes "linux" for _either_ x86 or arm. You may not even need to differentiate between `build-family: linux-aarch64` and `build-family: linux-x86_64` at all.

`build_linux_packages.sh` already chooses which Docker image to use, so no need to plumb that through: https://github.com/openxla/iree/commit/e791af1b82c59d278cb4837595c96fa67e477fcd

So to summarize, I think you only need to add
* `Main distribution (Linux-AArch64)` (if people are asking for it, we don't have macOS or Windows "main distributions", fwiw)
* `linux-aarch64` steps under the `matrix: include:` at the top of the file that use aarch64 runners to build the linux steps

Reply:
Fixed. I should have seen that :relieved: 

> if people are asking for it, we don't have macOS or Windows "main distributions", fwiw

I kept it for now. Is there a drawback to keep it?

Reply:
Possible drawbacks:

* Workflow file complexity
* Extra build time
* Potential for build instability (real build failures or flakes)

Reply:
Ok, seems to me like the drawbacks aren't that burdensome, but it might be me overlooking something. OTOH, if nobody is asking for it there's no point in doing it.... Unfortunately I don't have idea whether this is being asked for, so I rely on your judgement there. WDYT?